mongo/change_stream_test.go

```diff
@@ -546,3 +546,67 @@ func TestChangeStream_ReplicaSet(t *testing.T) {
 	// But, killCursors was already used to cause an error for the ResumeOnce test, so this does not need to be tested
 	// again.
 }
+
+func TestChangeStream_RegistryCheckInDecode(t *testing.T) {
```

Comment:
These test would serve the purpose and would fail before the code change in this PR.